Adds aesthetic basalt floor tiles

FantasticFwoosh opened the pull request:
:cl: Moonlighting Mac says
add: Courtesy of the art & crafts division, imitation basalt tiles themed on the rough volcanic terrain of lavaland are now available to be made out of sandstone.
experiment: The manufacturer even managed to replicate the way that it lights up with volcanic energy, it is the perfect accompaniment to other fake (or real) lava-land tiles or a independent piece.
/:cl:

🎨 This new aesthetic  item will allow you to not only make your fake pits more believable but to also cover over holes in places such as lavaland with excess sandstone you have gathered to help maintain the aesthetic ambience of the area if you do not want to resort to using station tiles to cover over them. This item can be placed anywhere on or off station as you need it conveniently without ruining atmos like real lavaland turfs. 

![](http://puu.sh/vEffs/cd9427004c.jpg)

(PR turf on the right, compared to grass)

FantasticFwoosh committed:
Adds fake basalt floor tile to sandstone recipies

Without meaning to split sand and ash (since they are not distinct enough) it is fair to add it to the relevant recipe of sandstone, mainly since you can dig up the turf the patch it over again

FantasticFwoosh committed:
Adds fake basalt tiles

FantasticFwoosh committed:
Adds fake basalt floors

Non-atmos set fake basalt floors for aeshetic purposes to compliment fake chasm without dying of oxyloss actually visiting lavaland- dig up to reclaim ash functionally the same as grass & place/pick up tiles. Also cycles sprites with realistic basalt lighting effect.

FantasticFwoosh committed:
Delete tiles.dmi

To resubmit

FantasticFwoosh committed:
Adds basalt tile sprite

FantasticFwoosh committed:
Update fancy_floor.dm

adds a return to allow it to hopefully compile now.

RandomMarine commented on code/game/turfs/simulated/floor/fancy_floor.dm:
```diff
@@ -113,6 +114,32 @@
+		icon_state = "basalt[rand(0, 12)]"
+		set_basalt_light(src)
+
+/turf/open/floor/grass/fakebasalt/attackby(obj/item/C, mob/user, params)
```
I don't think this is necessary since it inherits from grass turfs.

FantasticFwoosh replied:
Which explicitly? I don't really care to change it unless its a problem, since you can't dig up grass/snow/basalt turfs it made sense, and im still a amateur coder learning the ropes so im not sure it would have carried over correctly so having the grass interaction explicitly mentioned again was worthwhile just for my peace of mind.

Its copy-paste-modify code and it works as intended from playtesting in spitting out 2 ash with the spade interaction. I don't mind the extra work.

RandomMarine replied:
This whole proc, it's exactly the same as what grass uses (which this inherits) except that "uproot" is changed to "dig up". You should axe this proc and just change grass's to say "dig up" instead, since it's more accurate to what's being done (Especially for snow, too)

FantasticFwoosh replied:
But you ARE uprooting the grass in the tile (its a organic thing) but the ash tile doesn't have any organic content, if anybody wants to PR over that later be my guest.

Yeah, i know its the same proc, i copy pasted it for **_peace of mind_**, i literally do not mind nor care and i think just changing the description is enough since it works & compiles. Its really not worth worrying about since i don't think anybody will inherit from this and instead deviate to call upon the grass directly, and if people want to add more ores (the sand content dropped) they can.

> fake/real snow & ice tiling are things for later.

MMMiracles replied:
Just replace the "uproot" part with a var named something like 'turfverb' and use it on to give each turf a more appropriate description.

FantasticFwoosh replied:
That's a good idea but surely someone else's pr.

MMMiracles replied:
Have fun getting scolded for needless copypasta, then.

FantasticFwoosh replied:
Jokes aside, to express a point besides grass this is the only other turf to work in this way specifically so its something that can be done later when more tiles of this type are added as to make the whole system consistent.

optimumtact replied:
copypasted code, please refactor so that this isn't necessary

Niknakflak commented:
No screenshots

FantasticFwoosh commented:
Saying please would have been more polite @Niknakflak , this was taken during testing. I am dressed up to express my coding wizardry in bringing this feature too you. Also expresses my claim in self lighting up tiles.

![](https://puu.sh/vCZge/f3b43df3e0.jpg)

YakumoChen commented:
why not just post the sprite as in on the .dmm like literally everyone else

FantasticFwoosh commented:
❓

To **_try_** and answer your question, because its not a sprite used on any known map file. Its identical to lavaland turfs while being completely safe atmos-wise with the mechanics of grass tiles in being easy to place & dig up again with a shovelling object (or a lever)

> Trust me you should have seen my failed playtests where placing this tile caused all the air to rush out of the room, then choke you to death before fixed it.

> Everyone is different, no two people are the same

YakumoChen commented:
I mean the tile sprite, not the on-floor sprite

FantasticFwoosh commented:
Its in the tiles.dmi commit, you're never going to see it in that much detail anyway, so my screenshot is taken from in-game to kind of leverage what to expect.

Niknakflak commented:
That screenshot was so difficult to see you might as well not have posted anything.
Screenshots with sprites in PRs should be expected, nobody should have to say please.

FantasticFwoosh commented:
I was only jostling as a joke, i would have complied anyway (and i did). The lights were turned off as to illustrate the point of self lighting tiles (far left slightly lighting up the tactically placed statue as opposed to the flare), the same ambience as you experience on lava-land.

![](http://puu.sh/vDxXb/f336650ffe.jpg)

Here you go a better more lit up picture of boring sprites you've already seen before but now packaged much more safely.

optimumtact reviewed (changes requested):
Has some code that is needlessy copypasted from the parent turf

FantasticFwoosh committed:
Adds a turfverb var, cuts on copypasta

Thanks for the advice MMMiracles, anyway this [src]'s the turf you are going to dig up with appropriate verb via [src.turfverb]. Also removed copypasta code i was silly enough to use in its place. Its been playtested and it inherits & executes correctly. Some minor space shifting too.
